frontend/src/components/dashboard/PremiumOnboarding.tsx

```diff
@@ -384,21 +388,69 @@ const StepThree = () => {
   const { l10n } = useLocalization();
   const isLargeScreen = useMinViewportWidth("md");
 
+  let addonDescription = null;
+  let title = l10n.getString("multi-part-onboarding-reply-headline");
```

Comment:
`let` and `if/else if` smells bad to me. So I took a stab at refactoring this into new more-declarative components [here](https://github.com/mozilla/fx-private-relay/pull/1708). (And I learned a lot about React and TypeScript on the way!)

Reply:
Love that you're jumping into the React code! Left a review there, overall approach looks good :)